azure/services/virtualmachines/spec.go

```diff
@@ -196,6 +196,20 @@ func (s *VMSpec) generateStorageProfile() (*compute.StorageProfile, error) {
 		if s.OSDisk.ManagedDisk.DiskEncryptionSet != nil {
 			storageProfile.OsDisk.ManagedDisk.DiskEncryptionSet = &compute.DiskEncryptionSetParameters{ID: pointer.String(s.OSDisk.ManagedDisk.DiskEncryptionSet.ID)}
 		}
+		if s.OSDisk.ManagedDisk.SecurityProfile != nil {
+			if _, exists := s.SKU.GetCapability(resourceskus.ConfidentialComputingType); !exists {
+				return nil, azure.WithTerminalError(fmt.Errorf("vm size %s does not support confidential computing. select a different vm size or remove the security profile of the OS disk", s.Size))
```

Comment:
```suggestion
				return nil, azure.WithTerminalError(fmt.Errorf("VM type %s does not support confidential computing. Select a different VM type or remove the security profile of the OS disk", s.Size))
```

We seem to be using "type" more consistently in the rest of this PR, and I slightly prefer that to "size," if we don't want to use "sku."